backend/windmill-common/src/utils.rs

```diff
@@ -261,30 +269,43 @@ pub async fn report_critical_error(error_message: String, _db: DB) -> () {
     }
 
     #[cfg(feature = "enterprise")]
-    send_critical_alert(error_message, &_db, CriticalAlertKind::CriticalError, None).await;
+    if !*CLOUD_HOSTED {
+        send_critical_alert(error_message, &_db, CriticalAlertKind::CriticalError, None).await;
+    } else {
+        tracing::error!(error_message)
+    }
 }
 
-pub async fn report_recovered_critical_error(message: String, _db: DB) -> () {
+pub async fn report_recovered_critical_error(
+    message: String,
+    _db: DB,
+    workspace_id: Option<&str>
+) -> () {
     tracing::info!("RECOVERED CRITICAL ERROR: {message}");
 
     let mute = CRITICAL_ALERT_MUTE_UI_ENABLED.load(Ordering::Relaxed);
 
     if let Err(err) = sqlx::query!(
-        "INSERT INTO alerts (alert_type, message, acknowledged) VALUES ('recovered_critical_error', $1, $2)",
+        "INSERT INTO alerts (alert_type, message, acknowledged, workspace_id) VALUES ('recovered_critical_error', $1, $2, $3)",
         message,
-        mute
+        mute,
+        workspace_id,
     )
     .execute(&_db)
     .await
     {
         tracing::error!("Failed to save critical error to database: {}", err);
     }
     #[cfg(feature = "enterprise")]
+    if !*CLOUD_HOSTED {
```

Comment:
@rubenfiszel @HugoCasa  ☝️  never reporting to critical alerts channels if `CLOUD_HOSTED`